assembly/pom.xml

```diff
@@ -74,6 +74,11 @@
       <artifactId>spark-repl_${scala.binary.version}</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
```

Comment:
> So the end users do not have to specify `--packages` when they start the Spark Connect server.

What I mean is, if we are just trying to address this issue, it would be sufficient to just modify this file, and it appears that there is no need to move the code of the connect server module to a new directory.

Reply:
I will reply in the dev mailing list

Reply:
Maybe We can add the following `exclusions` to `spark-connect_${scala.binary.version`, which will prevent any changes to `spark-deps-hadoop-3-hive-2.3`, since the current `spark-connect` module is still a shaded jar and does not require the addition of extra third-party dependencies to the project:


```
<exclusions>
        <exclusion>
          <groupId>org.apache.spark</groupId>
          <artifactId>spark-connect-common_${scala.binary.version}</artifactId>
        </exclusion>
        <exclusion>
          <groupId>io.grpc</groupId>
          <artifactId>*</artifactId>
        </exclusion>
        <exclusion>
          <groupId>com.google.code.gson</groupId>
          <artifactId>gson</artifactId>
        </exclusion>
        <exclusion>
          <groupId>com.google.guava</groupId>
          <artifactId>failureaccess</artifactId>
        </exclusion>
      </exclusions>
```

Reply:
FYI: we are already good to unify the Guava in the whole Spark project, see https://github.com/apache/spark/pull/42493